src/RunScriptCommand.cs

```diff
@@ -47,16 +44,11 @@ public async Task<int> InvokeAsync(InvocationContext context)
         console.AlertAboutVerbose();
 
         var scriptShell = context.ParseResult.GetValueForOption(GlobalOptions.ScriptShell);
-        if (scriptShell is not null)
-        {
-            _projectScriptshell = scriptShell;
-        }
-
-        var (shell, isCmd) = GetScriptShell();
+        var (shell, isCmd) = GetScriptShell(scriptShell ?? _project.ScriptShell);
 
         var ct = context.GetCancellationToken();
 
-        foreach (var script in _scriptNames.Where(scriptName => _scripts.ContainsKey(scriptName)).ToImmutableArray())
+        foreach (var script in _scriptNames.Where(scriptName => _project.Scripts!.ContainsKey(scriptName)).ToImmutableArray())
```

Comment:
```suggestion
        foreach (var script in _scriptNames.Where(scriptName => _project.Scripts!.ContainsKey(scriptName)))
```

Since this was inlined the `.ToImmutableArray()` call isn't needed.